crates/bindings-csharp/Runtime/Internal/Bounds.cs

```diff
@@ -0,0 +1,305 @@
+        w.Write((byte)0);
+        new ZRW().Write(w, b.z.Max);
+    }
+}
+
+public readonly struct BTreeIndexBounds<T, TRW, U, URW, V, VRW, W, WRW, X, XRW, Y, YRW, Z, ZRW, A, ARW>
+    ((T t, U u, V v, W w, X x, Y y, Z z, Bound<A> a) b) : IBTreeIndexBounds
+    where A : IEquatable<A>
+    where TRW : struct, IReadWrite<T>
+    where URW : struct, IReadWrite<U>
+    where VRW : struct, IReadWrite<V>
+    where WRW : struct, IReadWrite<W>
+    where XRW : struct, IReadWrite<X>
+    where YRW : struct, IReadWrite<Y>
+    where ZRW : struct, IReadWrite<Z>
+    where ARW : struct, IReadWrite<A> {
+    public ushort PrefixElems => 7;
+
+    public void Prefix(BinaryWriter w) {
+        new TRW().Write(w, b.t);
+        new URW().Write(w, b.u);
+        new VRW().Write(w, b.v);
+        new WRW().Write(w, b.w);
+        new XRW().Write(w, b.x);
+        new YRW().Write(w, b.y);
+        new ZRW().Write(w, b.z);
+    }
+
+    public void RStart(BinaryWriter w) {
+        w.Write((byte)0);
+        new ARW().Write(w, b.a.Min);
+    }
+
+    public void REnd(BinaryWriter w) {
+        w.Write((byte)0);
+        new ARW().Write(w, b.a.Max);
+    }
+}
+
+public readonly struct BTreeIndexBounds<T, TRW, U, URW, V, VRW, W, WRW, X, XRW, Y, YRW, Z, ZRW, A, ARW, B, BRW>
+    ((T t, U u, V v, W w, X x, Y y, Z z, A a, Bound<B> b) b) : IBTreeIndexBounds
+    where B : IEquatable<B>
+    where TRW : struct, IReadWrite<T>
+    where URW : struct, IReadWrite<U>
+    where VRW : struct, IReadWrite<V>
+    where WRW : struct, IReadWrite<W>
+    where XRW : struct, IReadWrite<X>
+    where YRW : struct, IReadWrite<Y>
+    where ZRW : struct, IReadWrite<Z>
+    where ARW : struct, IReadWrite<A>
+    where BRW : struct, IReadWrite<B> {
+    public ushort PrefixElems => 8;
+
+    public void Prefix(BinaryWriter w) {
+        new TRW().Write(w, b.t);
+        new URW().Write(w, b.u);
+        new VRW().Write(w, b.v);
+        new WRW().Write(w, b.w);
+        new XRW().Write(w, b.x);
+        new YRW().Write(w, b.y);
+        new ZRW().Write(w, b.z);
+        new ARW().Write(w, b.a);
+    }
+
+    public void RStart(BinaryWriter w) {
+        w.Write((byte)0);
+        new BRW().Write(w, b.b.Min);
+    }
+
+    public void REnd(BinaryWriter w) {
+        w.Write((byte)0);
+        new BRW().Write(w, b.b.Max);
+    }
+}
+
+public readonly struct BTreeIndexBounds<T, TRW, U, URW, V, VRW, W, WRW, X, XRW, Y, YRW, Z, ZRW, A, ARW, B, BRW, C, CRW>
+    ((T t, U u, V v, W w, X x, Y y, Z z, A a, B b, Bound<C> c) b) : IBTreeIndexBounds
+    where C : IEquatable<C>
+    where TRW : struct, IReadWrite<T>
+    where URW : struct, IReadWrite<U>
+    where VRW : struct, IReadWrite<V>
+    where WRW : struct, IReadWrite<W>
+    where XRW : struct, IReadWrite<X>
+    where YRW : struct, IReadWrite<Y>
+    where ZRW : struct, IReadWrite<Z>
+    where ARW : struct, IReadWrite<A>
+    where BRW : struct, IReadWrite<B>
+    where CRW : struct, IReadWrite<C> {
+    public ushort PrefixElems => 9;
```

Comment:
Oh my god. This is getting out of hand 😅

I suppose we can fix it later though.

Reply:
yeah hehe